### PR TITLE
Nixos virtualbox module, additional options

### PR DESCRIPTION
**Pull request by joelmo**

### CODE REVIEW

**Comment by joelmo** [comment]
Btw, should double-quotes automatically be escaped with backslash?

File: nixos/modules/virtualisation/virtualbox-image.nix [hunk 1]
```diff
@@ -18,6 +18,23 @@ in {
...
+      baseImagePreUnmountHook = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Run commads after generating the GRUB menu and before unmounting
+          /mnt/{proc,dev,sys,}. This can be used to provide additional files in
+          the created image.
+        '';
+      };
+      OVApreExportHook = mkOption {
```
Comment (edolstra) [inline nixos/modules/virtualisation/virtualbox-image.nix]: Should be `preExportOVAHook`.

File: nixos/modules/virtualisation/virtualbox-image.nix [hunk 2]
```diff
@@ -18,6 +18,23 @@ in {
           a format the qemu-img command accepts.
         '';
       };
+      baseImagePreUnmountHook = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Run commads after generating the GRUB menu and before unmounting
```
Comment (edolstra) [inline nixos/modules/virtualisation/virtualbox-image.nix]: Type: `commads`

File: nixos/modules/virtualisation/virtualbox-image.nix [hunk 3]
```diff
@@ -18,6 +18,23 @@ in {
           a format the qemu-img command accepts.
         '';
       };
+      baseImagePreUnmountHook = mkOption {
+        type = types.str;
```
Comment (edolstra) [inline nixos/modules/virtualisation/virtualbox-image.nix]: This should be `types.lines`.

File: nixos/modules/virtualisation/virtualbox-image.nix [hunk 4]
```diff
@@ -18,6 +18,23 @@ in {
...
+        default = "";
+        description = ''
+          Run commads after generating the GRUB menu and before unmounting
+          /mnt/{proc,dev,sys,}. This can be used to provide additional files in
+          the created image.
+        '';
+      };
+      OVApreExportHook = mkOption {
+        type = types.str;
```
Comment (edolstra) [inline nixos/modules/virtualisation/virtualbox-image.nix]: Also `types.lines`.

File: nixos/modules/virtualisation/virtualbox-image.nix [hunk 5]
```diff
@@ -18,6 +18,23 @@ in {
...
+        default = "";
+        description = ''
+          Additional commands to run when creating the Virtualbox VM, these
```
Comment (edolstra) [inline nixos/modules/virtualisation/virtualbox-image.nix]: "VM, these" -> "VM. These"

File: nixos/modules/virtualisation/virtualbox-image.nix [hunk 6]
```diff
@@ -92,7 +109,9 @@ in {
             # Generate the GRUB menu.
             ln -s vda /dev/sda
             chroot /mnt ${config.system.build.toplevel}/bin/switch-to-configuration boot
-  
+            
+            eval "${config.virtualbox.baseImagePreUnmountHook}"
```
Comment (edolstra) [inline nixos/modules/virtualisation/virtualbox-image.nix]: Why `eval`?

**Comment by joelmo** [comment]
@edolstra thanks for reviewing. All should be fixed now, can you take a quick look?

**Comment by joelmo** [comment]
I want to use this to make some parts of /home available in the final image. But I have trouble to get this to work, I suspect the build-user can't see it. I have passed this additional option to QEMU `-virtfs local,path=/home/,security_model=none,mount_tag=home`. Any ideas how I can make this work?

**Pull request closed without merging**